scripts/prepDataForGephi.py

```diff
@@ -0,0 +1,330 @@
+        indexforSnpName = None
+        indexforPValue = None
+
+        snpsForThisPhecodeList = []
+        snpsForThisPhecodeSet = set()
+
+        # Figure out where phenotype name, snp name, and p-value are in the filtered data that are being processed
+        firstLine = lines[0]
+        firstLineAsArray = firstLine.split(' ')
+        for i in range(0, len(firstLineAsArray)):
+            if(firstLineAsArray[i] == phecodeColName):
+                indexforPhenotype = i
+            elif(firstLineAsArray[i] == snpColName):
+                indexforSnpName = i
+            elif(firstLineAsArray[i] == pValueColName):
+                indexforPValue = i
+
+        for elem in lines:
+            # Get the name of the phenotype
+            phecode = elem.split(fileDelimiter)[indexforPhenotype]
+            # Get the name and p-value of the SNP
+            snpName = elem.split(fileDelimiter)[indexforSnpName]
+            # If this isn't the first row of the file, add this tuple to our list/set
+            if(snpName != snpColName):
+                snpsForThisPhecodeSet.add(snpName)
+
+            if(indexforPValue != None):
+                snpPValue = elem.split(fileDelimiter)[indexforPValue]
+
+                # If this isn't the first row of the file, add this tuple to our list/set
+                if(snpName != snpColName):
+                    snpPValueFloat = float(snpPValue)
+                    snpsForThisPhecodeList.append((snpName, snpPValueFloat))
+        
+        # If SNPs have been found for this phecode, we add it to our mappings
+        if(len(snpsForThisPhecodeSet) > 0):
+            # Set is used for the edge map, to find overlaps of associations between phenotypes;
+            snpMappingsSet[phecode] = snpsForThisPhecodeSet
+
+        if(len(snpsForThisPhecodeList) > 0):
+        # List is used for the node map to present a sorted list of nodes
+            # Sort the list by p-value (i.e., by the second element of each tuple)
+            snpsForThisPhecodeList.sort(key = lambda x: float(x[1]), reverse = True)
+            snpsForThisPhecodeList.reverse()
+            # Get a list of the now sorted SNP IDs
+            snpNamesSorted = [i[0] for i in snpsForThisPhecodeList]
+            snpMappingsList[phecode] = snpNamesSorted
+
+print("Finished identifying phenotype to SNP mappings.")
+
+
+
+# If we want to create a node map file
+if(nodeMapInputPath != "" and nodeMapOutputPath != ""):
+    print("Creating node map...")
+
+    # Figure out if we need the p-value sorted list or not
+    numNodesToProcess = -1
+    if(len(snpMappingsList.keys()) > 0):
+        numNodesToProcess = len(snpMappingsList.keys())
+    else:
+        numNodesToProcess = len(snpMappingsSet.keys())
+
+    print("We will be processing " + str(numNodesToProcess) + " phenotypes in total for our node map.")
+
+    # Read in the node map input and output files
+    currentNodeMap = csv.reader(open(nodeMapInputPath, 'r'))
```

Comment:
use 
with(open(nodeMapInputPath, 'r')) as inf:
 reader = csv.reader(inf)
for row in reader:

the with will take care of flushing and closing the file. In some systems the files produced without flush/close don't have last few bytes and are corrupt.

Reply:
Do I need to do this for the csv.writer line as well?